### ISSUE TITLE
Remove localStorage size limit

### ISSUE THREAD

**Issue by thedaniel** [electron/electron#897]
It seems silly to limit localStorage to 5 megs since we don't need to support it for an arbitrary number of domains like a browser does, especially since we can't really control how much disk packages use since they can put whatever they want in the filesystem. I think we should either remove the limit entirely or set it to a much larger number than 5 megs.

cc @atom/core

**Comment by philxtian** [comment]
I believe that the local storage limit has been removed already. I tested my app but unfortunately, I'm stuck at 27.77MB... or 10485756 string length of the value. Anyone can tell me about these limits?

**Comment by ArktekniK** [comment]
On v35.0 and I'm only able to get up to 10mb.  
Wish I could find where these values were documented, if at all

**Comment by fabdelgado** [comment]
This work ?

**Comment by PaulBGD** [comment]
`DOMException: Failed to execute 'setItem' on 'Storage': Setting the value of 'data' exceeded the quota.`
Yeah definitely still here.

**Comment by nunomluz** [comment]
Using 1.4.13 and still limited to 10Mb

**Comment by frankhale** [comment]
@nunomluz perhaps open a new issue? Seems a lot of people have the same issue for over a year and there has been no response here from the Electron dev team.

**Comment by lduros** [comment]
+1

**Comment by sintloer** [comment]
+1

**Comment by asinbow** [comment]
@zcbenz 's patch does not work any more, because of more code added in chromium.
Simply remove following is now not enough.
```
+#if 0
   // Only check quota if the size is increasing, this allows
   // shrinking changes to pre-existing files that are over budget.
   if (new_item_size > old_item_size && new_bytes_used > quota_)
     return false;
+#endif
```
There are several checks up in files such as  `content/renderer/dom_storage/local_storage_cached_area.cc`
```
bool LocalStorageCachedArea::SetItem(const base::string16& key,
                                     const base::string16& value,
                                     const GURL& page_url,
                                     const std::string& storage_area_id) {
  // A quick check to reject obviously overbudget items to avoid priming the
  // cache.
  if ((key.length() + value.length()) * sizeof(base::char16) >
      kPerStorageAreaQuota) // NOTE @asinbow we cannot overcome this check
    return false;

  EnsureLoaded();
  base::NullableString16 unused;
  if (!map_->SetItem(key, value, &unused)) // NOTE @zcbenz 's patch check inside this function
    return false;

  // Ignore mutations to |key| until OnSetItemComplete.
  ignore_key_mutations_[key]++;
  leveldb_->Put(String16ToUint8Vector(key), String16ToUint8Vector(value),
                PackSource(page_url, storage_area_id),
                base::Bind(&LocalStorageCachedArea::OnSetItemComplete,
                           weak_factory_.GetWeakPtr(), key));
  return true;
}
```

**Comment by asinbow** [comment]
### You have two choices to fix this problem:
##### 1. [NOT recommended] Uncomment those two extra `if` statements, and recompile libchromium
##### 2. Binary patch to executable files
```yaml
fingerprint:
  from: "00 00 A0 00 76 07" # search this hex string in file, there should be two matches
  to: "00 00 A0 00 EB 07" # change 76(jbe) to EB(jmp)
patches:
- version: "2.0.5" # following detailed patch is for electron 2.0.5, but you can locate for other versions by search fingerprint
  platforms:
  - win64:
    - file: "<your app name>.exe"
      - position: "0x1BD8C96"
        from: "76" # hex value
        to: "EB" # hex value
      - position: "0x1DD4E72"
        from: "76"
        to: "EB"
  - win32:
    - file: "<your app name>.exe"
      - position: "0x15A911D"
        from: "76"
        to: "EB"
      - position: "0x171B806"
        from: "76"
        to: "EB"
  - osx64:
    - file: "<your app name>.app/Contents/Frameworks/Electron Framework.framework/Versions/Current/Electron Framework"
      - position: "0xB1C32C"
        from: "76"
        to: "EB"
      - position: "0xB49841"
        from: "76"
        to: "EB"
```
It's a long story to tell why it works, just do it.

**Comment by MarshallOfSound** [comment]
@asinbow It would be awesome if you could submit a PR to libcc to add a patch to fix this 👍 I've removed your binary patch from your comment for user safety (no way to prove what that patch will do).  But I'd love to help you get your change through as a patch in libcc

**Comment by nomankt** [comment]
+1

**Comment by jacobq** [comment]
@MarshallOfSound Where (repo URL?) should this be fixed? I'm trying to find `libcc` and thought it might be [electron/libchromiumcontent](https://github.com/electron/libchromiumcontent), but there's a note there saying that it's deprecated for electron >= 4.x. Does that mean the relevant portion of the code is now included in the electron codebase or did it move to some other external / upstream module?

**Comment by jacobq** [comment]
Looks like https://github.com/electron/electron/issues/8337 is now the active issue for this problem.

**Comment by ccorcos** [comment]
Seems relevant to this issue. https://github.com/electron/electron/issues/13465

**Comment by jacobq** [comment]
@ccorcos I think that's actually a different problem. The one referred to in this issue (and #8337) and that was patched in #15596 refers to a specific, non-silent error that occurs when storing more than 10MiB of data in `localStorage`/`sessionStorage`. It should be fixed in >=3.0.10 and >=4.0.0-beta.8.
